content/blog/deploying-mongodb-application-on-hpe-greenlake-for-containers.md

```diff
@@ -0,0 +1,948 @@
+Labels:                   app=mongo
+                          controller-revision-hash=mongo-7648bd99c8
+                          hpecp.hpe.com/hpecp-internal-gateway=true
+                          statefulset.kubernetes.io/pod-name=mongo-2
+Annotations:              hpecp-internal-gateway/27017: epicgw.customer.hpe.net:10035
+Selector:                 app=mongo,controller-revision-hash=mongo-7648bd99c8,statefulset.kubernetes.io/pod-name=mongo-2
+Type:                     NodePort
+IP Families:              <none>
+IP:                       10.101.219.63
+IPs:                      10.101.219.63
+Port:                     <unset>  27017/TCP
+TargetPort:               27017/TCP
+NodePort:                 <unset>  31216/TCP
+Endpoints:                10.192.4.208:27017
+Session Affinity:         None
+External Traffic Policy:  Cluster
+Events:
+  Type    Reason  Age   From         Message
+  ----    ------  ----  ----         -------
+  Normal  HpeCp   23m   hpecp-agent  Created HPECP K8S service
+
+
+
+ 
+Execute Mongo Shell into any one of the mongo services to configure Mongo Replica-Set
+
+Install “mongosh” client locally for shell interaction with mongodb. Mongosh will be accessing the mongo cluster from outside the Kubernetes cluster.
+Package : “mongosh-1.6.0-win32-x64.zip”
+$ export PATH=$PATH:/c/Ashish/mongosh-1.6.0-win32-x64/bin/
+
+
+$ mongosh --host epicgw.customer.hpe.net --port 10030
+Current Mongosh Log ID: 637f7622aaa80bc199cfcb06
+Connecting to:          mongodb://epicgw.customer.hpe.net:10030/?directConnection=true&appName=mongosh+1.6.0
+Using MongoDB:          6.0.3
+Using Mongosh:          1.6.0
+
+For mongosh info see: https://docs.mongodb.com/mongodb-shell/
+
+------
+   The server generated these startup warnings when booting
+   2022-11-24T12:48:39.726+00:00: Access control is not enabled for the database. Read and write access to data and configuration is unrestricted
+   2022-11-24T12:48:39.727+00:00: You are running this process as the root user, which is not recommended
+   2022-11-24T12:48:39.728+00:00: You are running on a NUMA machine. We suggest launching mongod like this to avoid performance problems: numactl --interleave=all mongod [other options]
+   2022-11-24T12:48:39.728+00:00: vm.max_map_count is too low
+------
+
+------
+   Enable MongoDB's free cloud-based monitoring service, which will then receive and display
+   metrics about your deployment (disk utilization, CPU, operation statistics, etc).
+
+   The monitoring data will be available on a MongoDB website with a unique URL accessible to you
+   and anyone you share the URL with. MongoDB may use this information to make product
+   improvements and to suggest MongoDB products and deployment options to you.
+
+   To enable free monitoring, run the following command: db.enableFreeMonitoring()
+   To permanently disable this reminder, run the following command: db.disableFreeMonitoring()
+------
+
+test>
+
+Initialize Replicaset
```

Comment:
Oh... maybe not, because I now see this one as similar to the one above. Still, it almost feels like another Step- and seems to need a little more explanation of what follows.

Reply:
done